tests/unit/test_charm.py

```diff
@@ -58,8 +58,67 @@ def test_on_config_changes_aborts_if_host_is_container(harness, mocker):
     assert not harness.charm._stored.started
 
 
+def test_get_initiator_name_from_file(harness, mocker, tmp_path):
+    """Test fetching the initiator name from initiatorname.iscsi file."""
+    initiator_content = dedent(
+        """\
+        ###############################################################################
+        # [ WARNING ]
+        # configuration file maintained by Juju
+        # local changes will be overwritten.
+        #
+        # DO NOT EDIT OR REMOVE THIS FILE!
+        # If you remove this file, the iSCSI daemon will not start.
+        # If you change the InitiatorName, existing access control lists
+        # may reject this initiator.  The InitiatorName must be unique
+        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
+        ###############################################################################
+        InitiatorName=test-iqn"""
+    )
+
+    wrong_initiator_content = dedent(
+        """\
+        ###############################################################################
+        # [ WARNING ]
+        # configuration file maintained by Juju
+        # local changes will be overwritten.
+        #
+        # DO NOT EDIT OR REMOVE THIS FILE!
+        # If you remove this file, the iSCSI daemon will not start.
+        # If you change the InitiatorName, existing access control lists
+        # may reject this initiator.  The InitiatorName must be unique
+        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
+        ###############################################################################
+        wrongName=test-iqn"""
+    )
+
+    # case 1: File not present
+    initiatorname_file = tmp_path / "initiatorname.iscsi"
+    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
+    assert initiator_name is None
+
+    # case 2: File with empty contents
+    initiatorname_file.write_text("")
+    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
+    assert initiator_name is None
+
+    # case 3: File with wrong format of "initiatorName"
+    initiatorname_file.write_text(wrong_initiator_content)
+    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
+    assert initiator_name is None
+
+    # case 4: File with proper contents
```

Comment:
**nitpick (non-blocking):** 

Can we separate those case in different functions?

Reply:
I think you can use  `parameterized` again.